sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -1235,48 +1255,72 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
       (c, evPrim, evNull) => code"$evPrim = (long) $c;"
   }
 
-  private[this] def castToFloatCode(from: DataType): CastFunction = from match {
-    case StringType =>
-      (c, evPrim, evNull) =>
-        code"""
+  private[this] def castToFloatCode(from: DataType, ctx: CodegenContext): CastFunction = {
+    from match {
+      case StringType =>
+        val str = ctx.freshVariable("str", StringType)
+        (c, evPrim, evNull) =>
+          code"""
           try {
             $evPrim = Float.valueOf($c.toString());
           } catch (java.lang.NumberFormatException e) {
-            $evNull = true;
+              final String $str = $c.toString().trim();
```

Comment:
indentation?

Reply:
Although this is called only in case of exceptions, it would be great to call `.toString` once by reusing the one in line 1265.

Reply:
@dongjoon-hyun oops.. sorry.. i will fix.